ShellScript/Bash.sublime-syntax

```diff
@@ -1332,12 +1333,11 @@ contexts:
     - match: ==?|!=?|<|>|\|\||&&
       scope: keyword.operator.logical.shell
 
-  maybe-operator-logical:
-    - match: \!
+  operator-exclamation:
+    - match: \!(?=\s)
       scope: keyword.operator.logical.shell
-      pop: true
-    - match: $|(?=\S)
-      pop: true
+    - match: \!(?=\S)
```

Comment:
Maybe match the entire token as something in `variable` or `support`? This is only really relevant for an interactive shell anyway (with e.g. Terminus), but we should still try our best there, imo.

Reply:
You mean including the history command like  `!command`?

Reply:
... not sure whether it works well as `!` can be followed by expression.

e.g.: `![[ $1 == '0' ]] && echo -e "$2 PASSED" || echo -e "$2 FAILED";`

Reply:
We could enclose everything in a `meta.history.shell`?

```
  operator-exclamation:
    - match: \!(?!\S)
      scope: keyword.operator.logical.shell
    - match: \!
      scope: punctuation.definition.history.shell
      push:
        - - meta_scope: meta.history.shell
          - match: ''
            pop: true
        - cmd
```

Reply:
Tbh, I've never used this feature and shallow description from the net didn't exactly enlighten me. I know there are `!!` and `!<n>` short forms, but that's about it. If you don't have an idea to implement this correctly, just leave it as-is, imo.

Reply:
`!!` is what I didn't find in http://tldp.org/LDP/abs/html/special-chars.html#NOTREF

Both of your examples are references to the history index and could therefore be handled in a dedicated rule.